### PR TITLE
Return value in sigaction

### PR DESCRIPTION
**Pull request by danghai**
Hi Nick, 

I add the return handler in sigaction(). 
If unsuccessful, no new signal handler is installed, sigaction() returns -1, and sets errno to one of the following values

### CODE REVIEW

File: src/tlog-play.c [hunk 1]
```diff
@@ -451,7 +451,12 @@ run(struct tlog_errs **perrs,
 
     /* Setup signal handlers to terminate gracefully */
     for (i = 0; i < TLOG_ARRAY_SIZE(exit_sig); i++) {
-        sigaction(exit_sig[i], NULL, &sa);
+        if(sigaction(exit_sig[i], NULL, &sa) == -1){
+          grc = TLOG_GRC_ERRNO;
+          tlog_errs_pushc(perrs, grc);
+          tlog_errs_pushs(perrs, "Failed sigaction");
```
Comment (spbnick) [inline src/tlog-play.c]: Please provide a more specific error message. Something like: "Failed to retrieve an exit signal action".

File: src/tlog-play.c [hunk 2]
```diff
@@ -460,7 +465,12 @@ run(struct tlog_errs **perrs,
             }
             /* NOTE: no SA_RESTART on purpose */
             sa.sa_flags = 0;
-            sigaction(exit_sig[i], &sa, NULL);
+            if(sigaction(exit_sig[i], &sa, NULL) == -1){
+              grc = TLOG_GRC_ERRNO;
+              tlog_errs_pushc(perrs, grc);
+              tlog_errs_pushs(perrs, "Failed sigaction");
```
Comment (spbnick) [inline src/tlog-play.c]: Same here. Something like: "Failed to set an exit signal action".

File: src/tlog-play.c [hunk 3]
```diff
@@ -470,7 +480,12 @@ run(struct tlog_errs **perrs,
     sigaddset(&sa.sa_mask, SIGIO);
     /* NOTE: no SA_RESTART on purpose */
     sa.sa_flags = 0;
-    sigaction(SIGIO, &sa, NULL);
+    if(sigaction(SIGIO, &sa, NULL) == -1){
+      grc = TLOG_GRC_ERRNO;
+      tlog_errs_pushc(perrs, grc);
+      tlog_errs_pushs(perrs, "Failed sigaction");
```
Comment (spbnick) [inline src/tlog-play.c]: Same here. Something like: "Failed to set SIGIO action".

**Comment by spbnick** [comment]
Oh, and please also provide a proper commit subject. Something like "Check sigaction return codes".

**Comment by spbnick** [comment]
BTW, is that your dog on your userpic?

**Comment by danghai** [comment]
Hi Nick, 
I already fixed all return code for sigaction code. I saw one thread that you want to use argv ? In the tlog_play, can I pause or you miss this function?  
Haha, you mean Corgi dog? It is not my dog; it is my roommate's dog.

File: src/tlog-play.c [hunk 4]
```diff
@@ -823,7 +823,12 @@ run(struct tlog_errs **perrs,
 
     /* Restore signal handlers */
     for (i = 0; i < TLOG_ARRAY_SIZE(exit_sig); i++) {
-        sigaction(exit_sig[i], NULL, &sa);
+        if(sigaction(exit_sig[i], NULL, &sa) == -1) {
+          grc = TLOG_GRC_ERRNO;
+          tlog_errs_pushc(perrs, grc);
+          tlog_errs_pushs(perrs, "Failed to retrieve an exit signal action");
+          goto cleanup;
```
Comment (spbnick) [inline src/tlog-play.c]: This shouldn't go to `cleanup`, it's already there. It should just add errors to the stack and get on with the cleanup.

**Comment by spbnick** [comment]
Ah, I see. Asking because we are planning to get a corgi ourselves, now that we moved into a house :)

**Comment by danghai** [comment]
Hi Nick, I squash 3 commit into one and check the sigaction return code.  
btw, you should raise the Corgi dog. Haha, they are smart and cute.

File: lib/rec.c [hunk 5]
```diff
@@ -864,7 +869,12 @@ tlog_rec_transfer(struct tlog_errs    **perrs,
             }
             /* NOTE: no SA_RESTART on purpose */
             sa.sa_flags = 0;
-            sigaction(exit_sig[i], &sa, NULL);
+            if(sigaction(exit_sig[i], &sa, NULL) == -1){
+                grc = TLOG_GRC_ERRNO;
+                tlog_errs_pushc(perrs, grc);
+                tlog_errs_pushs(perrs, "Failed to retrieve an exit signal action");
```
Comment (spbnick) [inline lib/rec.c]: This should say "set", not "retrieve".

File: lib/rec.c [hunk 6]
```diff
@@ -873,7 +883,12 @@ tlog_rec_transfer(struct tlog_errs    **perrs,
     sigemptyset(&sa.sa_mask);
     /* NOTE: no SA_RESTART on purpose */
     sa.sa_flags = 0;
-    sigaction(SIGALRM, &sa, NULL);
+    if(sigaction(SIGALRM, &sa, NULL) == -1){
+      grc = TLOG_GRC_ERRNO;
+      tlog_errs_pushc(perrs, grc);
+      tlog_errs_pushs(perrs, "Failed to retrieve an SIGALRM signal action");
```
Comment (spbnick) [inline lib/rec.c]: Same here.

**Comment by spbnick** [comment]
Ah, sorry, no, lib/tty_source.c doesn't need the sigaction error handled, so all is done now, except the two comments about error messages.

**Comment by spbnick** [comment]
Yes, smart and cute are exactly the qualities we're looking forward to :)

**Pull request closed without merging**